packages/mui-material-next/src/ButtonBase/ButtonBase.test.tsx

```diff
@@ -0,0 +1,1101 @@
+        expect(onDragEnd.callCount).to.equal(1);
+      }
+
+      fireEvent.mouseDown(button);
+      expect(onMouseDown.callCount).to.equal(1);
+
+      fireEvent.mouseUp(button);
+      expect(onMouseUp.callCount).to.equal(1);
+
+      fireEvent.contextMenu(button);
+      expect(onContextMenu.callCount).to.equal(1);
+
+      fireEvent.click(button);
+      expect(onClick.callCount).to.equal(1);
+
+      act(() => {
+        button.focus();
+      });
+      expect(onFocus.callCount).to.equal(1);
+
+      fireEvent.keyDown(button);
+      expect(onKeyDown.callCount).to.equal(1);
+
+      fireEvent.keyUp(button);
+      expect(onKeyUp.callCount).to.equal(1);
+
+      act(() => {
+        button.blur();
+      });
+      expect(onBlur.callCount).to.equal(1);
+
+      fireEvent.mouseLeave(button);
+      expect(onMouseLeave.callCount).to.equal(1);
+    });
+  });
+
+  describe('ripple', () => {
+    describe('interactions', () => {
+      it('should not have a focus ripple by default', () => {
+        const { getByRole } = render(
+          <ButtonBase
+            TouchRippleProps={{
+              classes: {
+                ripplePulsate: 'ripple-visible',
+              },
+            }}
+          />,
+        );
+        const button = getByRole('button');
+        simulatePointerDevice();
+
+        focusVisible(button);
+
+        expect(button.querySelectorAll('.ripple-visible')).to.have.lengthOf(0);
+      });
+
+      it('should start the ripple when the mouse is pressed', () => {
+        const { getByRole } = render(
+          <ButtonBase
+            TouchRippleProps={{
+              classes: {
+                rippleVisible: 'ripple-visible',
+                child: 'child',
+                childLeaving: 'child-leaving',
+              },
+            }}
+          />,
+        );
+        const button = getByRole('button');
+
+        fireEvent.mouseDown(button);
+
+        expect(button.querySelectorAll('.ripple-visible .child-leaving')).to.have.lengthOf(0);
+        expect(
+          button.querySelectorAll('.ripple-visible .child:not(.child-leaving)'),
+        ).to.have.lengthOf(1);
+      });
+
+      it('should stop the ripple when the mouse is released', () => {
+        const { getByRole } = render(
+          <ButtonBase
+            TouchRippleProps={{
+              classes: {
+                rippleVisible: 'ripple-visible',
+                child: 'child',
+                childLeaving: 'child-leaving',
+              },
+            }}
+          />,
+        );
+        const button = getByRole('button');
+        fireEvent.mouseDown(button);
+
+        fireEvent.mouseUp(button);
+
+        expect(button.querySelectorAll('.ripple-visible .child-leaving')).to.have.lengthOf(1);
+        expect(
+          button.querySelectorAll('.ripple-visible .child:not(.child-leaving)'),
+        ).to.have.lengthOf(0);
+      });
+
+      it('should start the ripple when the mouse is pressed 2', () => {
```

Comment:
This one looks like it's already covered by `it('should restart the ripple when the mouse is pressed again'` below 👀

Reply:
It looks like a long time ago the test component was rendered only once, and each test chained off the state of the previous test ([source](https://github.com/mui/material-ui/blob/f69797ffa61fa3ae94cf62987c197d6acea371c9/packages/material-ui/src/ButtonBase/ButtonBase.test.js#L142-L144))

Since then they were isolated in https://github.com/mui/material-ui/pull/17394

Reply:
Removed